scijava/scijava-ops-api/src/main/java/org/scijava/ops/api/OpRequest.java

```diff
@@ -132,4 +135,68 @@ static void append(final StringBuilder sb, final String s) {
 			sb.append("/");
 		sb.append(s);
 	}
+
+}
+
+class PartialOpRequest implements OpRequest {
+
+	private final Optional<String> name;
+	private final Optional<Type[]> args;
+	private final Optional<Type> outType;
```

Comment:
If we're just going to return `null` when these are empty, might as well just set them to `null`...